addons/delivery_shipper/models/delivery_carrier.py

```diff
@@ -0,0 +1,245 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from markupsafe import Markup
+from odoo.exceptions import UserError
+from .shipper_service import Shipper
+import logging
+import math
+
+logger = logging.getLogger(__name__)
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = 'delivery.carrier'
+
+    delivery_type = fields.Selection(selection_add=[
+        ('shipper', 'Shipper'),
+    ], ondelete={'shipper': lambda records: records.write({'delivery_type': 'fixed', 'fixed_price': 0})})
+
+    shipper_api_key = fields.Char(help='Shipper API Integration key')
+    shipper_default_package_type_id = fields.Many2one(
+        'stock.package.type',
+        string='Default Package Type for Shipper',
+        help='Package dimensions are required to get more accurate rates. You can define these in a package type that you set as default',
+    )
+    shipper_origin_address = fields.Many2one(
+        string="Origin Address",
+        help="This address will be used when fetching the available services from shipper.",
+        comodel_name='res.partner',
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
+        default=lambda self: self.env.company.partner_id,
+    )
+    shipper_3pl_partner = fields.Selection([
+        ('57', 'J&T - Regular'),
+        ('228', 'Ninja Xpress - Regular'),
+        ('349', 'SAP - Regular'),
+        ('58', 'SiCepat - Regular'),
+        ('562', 'Anteraja - Regular'),
+        ('1', 'JNE - Regular'),
+        ('564', 'Anteraja - Express'),
+        ('59', 'SiCepat - Express'),
+        ('2', 'JNE - Express'),
+    ], string="Shipper 3PL Partners")
+    shipper_3pl_partner_id = fields.Integer(compute='_compute_shipper_3pl_partner_id', store=True)
+
+    @api.depends('shipper_3pl_partner')
+    def _compute_shipper_3pl_partner_id(self):
+        for record in self:
+            record.shipper_3pl_partner_id = int(record.shipper_3pl_partner) if record.shipper_3pl_partner else 0
+
+    # Shipping Carrier Methods
+    # TODO: Add an option for using geolocation instead; maybe
+    def shipper_rate_shipment(self, order):
+        shipper = self._get_shipper()
+        origin_area_id = self._shipper_get_area_id(origin=self.shipper_origin_address)
```

Comment:
You won't be able to use the origin set on the carrier; you'll need to use the order warehouse.
Also, please make your method private (add _ at the beginning). This applies to each method not expected to be accessed by RPC.